comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -115,18 +115,33 @@ func (c *collector) Start(ctx context.Context, store workloadmeta.Component) err
 		log.Warnf("cannot determine ECS cluster name: %s", err)
 	}
 
-	c.setTaskCollectionParser()
+	c.setTaskCollectionParser(instance.Version)
 
 	return nil
 }
 
-func (c *collector) setTaskCollectionParser() {
-	_, err := ecsmeta.V4FromCurrentTask()
-	if c.taskCollectionEnabled && err == nil {
-		c.taskCollectionParser = c.parseTasksFromV4Endpoint
+func (c *collector) setTaskCollectionParser(version string) {
+	if !c.taskCollectionEnabled {
+		log.Infof("detailed task collection disabled, using metadata v1 endpoint")
+		c.taskCollectionParser = c.parseTasksFromV1Endpoint
 		return
 	}
-	c.taskCollectionParser = c.parseTasksFromV1Endpoint
+
+	ok, err := ecsmeta.IsMetadataV4Available(util.ParseECSAgentVersion(version))
+	if err != nil {
+		log.Warnf("detailed task collection enabled but agent cannot determine if v4 metadata endpoint is available, using metadata v1 endpoint: %s", err.Error())
```

Comment:
This warning isn’t 100% accurate because `err != nil` can mean two different things:
* Either `IsMetadataV4Available` hasn’t been able parse the version of the ECS agent.
  * This is really unexpected since the ECS agent version is extracted thanks to a regex that should have validated its format.
  * So, this case indeed deserves a warning.
* Or `IsMetadataV4Available` properly parsed the ECS agent version and it has determined that the ECS agent is too old to support V4.
  * In that case, nothing went wrong from the agent point of view. So, we shouldn’t emit a warning.
  * A debug log could potentially be useful. But not because `agent cannot determine if v4 metadata endpoint is available` but because `agent has determined that v4 metadata endpoint is unavailable`.